src/chatdDb.h

```diff
@@ -381,7 +381,15 @@ class ChatdSqliteDb: public chatd::DbInterface
         SqliteStmt stmt(mDb, "select min(idx) from history where chatid = ?");
         stmt << mChat.chatId();
         stmt.stepMustHaveData(__FUNCTION__);
-        return stmt.uint64Col(0);
+
+        if (sqlite3_column_type(stmt, 0) == SQLITE_NULL) // no db history
```

Comment:
I think this change may break the existing usage of this method to initialize the `idx` used to add new messages backwards in `msgIncoming()`. What is its motivation?

Reply:
Before this change the only usage for this function was in msgIncoming as you said but

Reply:
Changes undone